astropy/modeling/core.py

```diff
@@ -2687,14 +2686,16 @@ def render_model(model, arr=None, coords=None):
     if arr is not None:
         arr = arr.copy()
         # Check dimensions match model
-        assert arr.ndim == model.n_inputs
-
+        if arr.ndim != model.n_inputs:
+            raise AssertionError
```

Comment:
`ValueError('number of array dimensions inconsistent with number of model inputs')`